SuperheroSightings/src/main/java/org/ss/DAO/LocationDAOImpl.java

```diff
@@ -1,12 +1,109 @@
 package org.ss.DAO;
 
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+import org.ss.DTO.Location;
+
 /**
  *@author : Claude Seide, Everlyn Leon, Mariya Malakhava, Neyssa Cadet
  *
  */
 
+@Repository
+public class LocationDAOImpl implements LocationDAO {
+
+    private final JdbcTemplate jdbc;
+
+    @Autowired
+    public LocationDAOImpl(JdbcTemplate jdbc){
+        this.jdbc = jdbc;
+    }
+
+    @Override
+    public Location getLocationByID(int locationID) {
+        try {
+            final String SQL = "SELECT * FROM Locations WHERE locationID = ?";
+            return jdbc.queryForObject(SQL, new LocationMapper(), locationID);
+        } catch (DataAccessException ex) {
+            return null;
+        }
+    }
+
+    @Override
+    @Transactional
+    public Location addNewLocation(Location location) {
+        final String SQL = "INSERT INTO Locations(locationName, locationDescription, locationAddress, locationLatitude, locationLongitude" +
+            "VALUES(?,?,?,?,?)";
+            jdbc.update(SQL,
+            location.getLocationName(),
+            location.getLocationDescription(),
+            location.getLocationAddress(),
+            location.getLocationLatitude(),
+            location.getLocationLongitude());
+
+            int newID = jdbc.queryForObject(SQL, Integer.class);
+            location.setLocationID(newID);
+            return location;
+    }
+
+    @Override
+    public List<Location> getAllLocations() {
+        final String SQL = "SELECT * FROM Locations";
+        return jdbc.query(SQL, new LocationMapper());
+    }
+
+    @Override
+    public void updateLocation(Location location) {
+        final String SQL = "UPDATE Locations SET locationName = ?, locationDescription = ?, locationAddress = ?, locationLatitude = ?, locationLongitude = ? WHERE locationID = ?";
+        jdbc.update(SQL, 
+            location.getLocationName(),
+            location.getLocationDescription(),
+            location.getLocationAddress(),
+            location.getLocationLatitude(),
+            location.getLocationLongitude());
+    }
+
+    @Override
+    @Transactional
+    public void deleteLocation(int locationID) {
+        final String DELETE_LOCATION = "DELETE FROM Locations WHERE locationID =?";
+        jdbc.update(DELETE_LOCATION, locationID);
+
+        final String DELETE_SIGHTING = "DELETE FROM Sightings WHERE locationID = ?";
```

Comment:
But I do have a question, when I implement the delete method, do I go ahead and delete the location in every table the key is referenced in ?